python/cudf/cudf/core/column/column.py

```diff
@@ -487,42 +488,67 @@ def __setitem__(self, key, value):
 
         if is_scalar(value):
             if is_categorical_dtype(self.dtype):
-                from cudf.utils.cudautils import fill_value
-
-                data = rmm.device_array(nelem, dtype=self.codes.dtype)
-                fill_value(data, self._encode(value))
-                value = build_categorical_column(
-                    categories=self.dtype.categories,
-                    codes=as_column(data),
-                    ordered=self.dtype.ordered,
-                )
-            elif value is None:
-                value = column.column_empty(nelem, self.dtype, masked=True)
-            else:
-                to_dtype = pd.api.types.pandas_dtype(self.dtype)
-                value = utils.scalar_broadcast_to(value, nelem, to_dtype)
-
-        value = column.as_column(value).astype(self.dtype)
+                value = self._encode(value)
 
-        if len(value) != nelem:
-            msg = (
-                f"Size mismatch: cannot set value "
-                f"of size {len(value)} to indexing result of size "
-                f"{nelem}"
-            )
-            raise ValueError(msg)
+            value = self.dtype.type(value) if value is not None else value
```

Comment:
When is `value` None here? `._encode` should already return a strongly typed numpy scalar so this feels unnecessary.

Reply:
If I try to set None to a set of values, like sr[1:4] = None.

will change it to 
```
if is_categorical_dtype(self.dtype):
    value = self._encode(value)
else:
    value = self.dtype.type(value) if value is not None else value

```